### PR TITLE
ddl: Use snapshot to add index

### PR DESCRIPTION
**Pull request by zimulala**
Using job‘s snapshot version instead of every transaction's `startTS` when on the state of write-reorganization. It can reduce unnecessary to read and write some rows that are inserted in the write-reorganization state.

### CODE REVIEW

**Comment by zimulala** [comment]
PTAL @winkyao @coocood

**Comment by coocood** [comment]
Please add the description for this PR.

**Comment by zimulala** [comment]
Done. @coocood

File: ddl/index_change_test.go [hunk 1]
```diff
@@ -103,9 +105,20 @@ func (s *testIndexChangeSuite) TestIndexChange(c *C) {
...
+			}
+			err = s.addRecordWriteReorg(ctx1, writeReorgTable, writeReorgStartVal, writeReorgStep)
+			if err != nil {
+				checkErr = errors.Trace(err)
+			}
+			writeReorgStartVal += writeReorgStep
 		case model.StatePublic:
-			if job.GetRowCount() != 3 {
-				checkErr = errors.Errorf("job's row count %d != 3", job.GetRowCount())
+			reorgRows := 3 + int64(writeReorgStep)
```
Comment (coocood) [inline ddl/index_change_test.go]: Why row count is not 3?
If only 3 records exists before reorg, and use snapshot, job row count should be 3.

**Comment by zimulala** [comment]
PTAL @coocood

**Comment by zimulala** [comment]
PTAL @winkyao @breeswish

**Comment by coocood** [comment]
LGTM

**Comment by shenli** [comment]
Is this PR ready for review?

**Comment by zimulala** [comment]
This PR needs to adjust the adding tests.

**Pull request closed without merging**